core/sql/optimizer/NormRelExpr.cpp

```diff
@@ -2436,15 +2449,21 @@ void Join::createAFilterGrandChildIfNeeded(NormWA & normWARef)
 
       if ((doNotUnnest == FALSE) && candidateForLeftJoin && 
           (CmpCommon::getDefault(SUBQUERY_UNNESTING_P2) != DF_INTERNAL) &&
-          (normWARef.getLeftJoinConversionCount() >= 1))
+          ((normWARef.getLeftJoinConversionCount() >= 1)||nestedAggInSubQ))
       {
         doNotUnnest = TRUE;
         // For phase 2 we only unnest 1 level of subqueries 
         // containing NonNullRejecting Predicates
 
-        if (CmpCommon::getDefault(SUBQUERY_UNNESTING) == DF_DEBUG)
+        if (CmpCommon::getDefault(SUBQUERY_UNNESTING) == DF_DEBUG) 
+        {
+          if (!nestedAggInSubQ)
             *CmpCommon::diags() << DgSqlCode(2997)
               << DgString1("Skipping unnesting of Subquery due to NonNullRejecting Predicates in more than one subquery");
+          else
+             *CmpCommon::diags() << DgSqlCode(2997)
+              << DgString1("Skipping unnesting of Subquery since we have both NonNullRejecting predicate and nested nested aggregate in subquery.");
```

Comment:
I was trying to make the debug message more detailed. If it is nestedAgg case we get one message, while if we have several LeftJoin requiring subqueries we get a different message. I suppose if both conditions are true we will get the more than one (now two with new commit) LeftJoin subquery message.